### DISCUSSION TITLE
NumPy array: pointer to shape and stride? size?

### DISCUSSION THREAD

**Discussion by tdegeus** [wjakob/nanobind#158] [Q&A]
### Problem description

It would be great to have
```cpp
size_t ndarray::size
```
and to be able to use e.g. `std::copy` on `shape` and `stride`, i.e. to make their pointers somehow public.

Would this be considered?

**Comment by wjakob** [comment]
Issues are only for reporting actual issues of the library, please don't use them for feature requests or questions. I will convert this to a discussion.

**Reply by tdegeus** [reply]
Sorry, I overlook that. Thanks

**Reply by wjakob** [reply]
Regarding your question: yes, it sounds like a reasonable request. You could call the pointer versions `shape_ptr` and `strides_ptr`. Please feel free to create a PR.
